### ISSUE TITLE
Having issues running on Google Cloud Genomics

### ISSUE THREAD

**Issue by saliksyed** [google/deepvariant#116]
I'm attempting to launch DeepVariant on GCP, with the VerilyGRCH38 ref genome and the whole exome model. Unfortunately the job never finishes. Any pointers on how to debug this would be much appreciated.

```
#!/bin/bash
set -euo pipefail
# Set common settings.
PROJECT_ID=valis-194104
OUTPUT_BUCKET=gs://canis/CNR-data
STAGING_FOLDER_NAME=deep_variant_files
OUTPUT_FILE_NAME=TLE_a_001_deep_variant.vcf
# Model for calling whole exome sequencing data.
MODEL=gs://deepvariant/models/DeepVariant/0.7.0/DeepVariant-inception_v3-0.7.0+data-wes_standard
IMAGE_VERSION=0.7.0
DOCKER_IMAGE=gcr.io/deepvariant-docker/deepvariant:"${IMAGE_VERSION}"
COMMAND="/opt/deepvariant_runner/bin/gcp_deepvariant_runner \
  --project ${PROJECT_ID} \
  --zones us-west1-b \
  --docker_image ${DOCKER_IMAGE} \
  --outfile ${OUTPUT_BUCKET}/${OUTPUT_FILE_NAME} \
  --staging ${OUTPUT_BUCKET}/${STAGING_FOLDER_NAME} \
  --model ${MODEL} \
  --regions gs://canis/CNR-data/CDS-canonical.bed \
  --bam gs://canis/CNR-data/TLE_a_001.bam \
  --bai gs://canis/CNR-data/TLE_a_001.bam.bai \
  --ref gs://genomics-public-data/references/GRCh38_Verily/GRCh38_Verily_v1.genome.fa  \
  --gcsfuse"
# Run the pipeline.
gcloud alpha genomics pipelines run \
    --project "${PROJECT_ID}" \
    --service-account-scopes="https://www.googleapis.com/auth/cloud-platform" \
    --logging "${OUTPUT_BUCKET}/${STAGING_FOLDER_NAME}/runner_logs_$(date +%Y%m%d_%H%M%S).log" \
    --zones us-west1-b \
    --docker-image gcr.io/deepvariant-docker/deepvariant_runner:"${IMAGE_VERSION}" \
    --command-line "${COMMAND}"
```

I get the following error: 
```
      Traceback (most recent call last):
        File "/opt/deepvariant_runner/src/gcp_deepvariant_runner.py", line 862, in <module>
          run()
        File "/opt/deepvariant_runner/src/gcp_deepvariant_runner.py", line 845, in run
          _run_make_examples(pipeline_args)
        File "/opt/deepvariant_runner/src/gcp_deepvariant_runner.py", line 340, in _run_make_examples
          _wait_for_results(threads, results)
        File "/opt/deepvariant_runner/src/gcp_deepvariant_runner.py", line 352, in _wait_for_results
          result.get()
        File "/usr/lib/python2.7/multiprocessing/pool.py", line 572, in get
          raise self._value
      RuntimeError: Job failed with error "run": operation "projects/valis-194104/operations/13939489157244551677" failed: executing pipeline: Execution failed: action 5: unexpected exit status 1 was not ignored (reason: FAILED_PRECONDITION)
    details:
```

**Comment by AndrewCarroll** [comment]
Hello Salik,

I am sorry you are encountering issues with the Cloud Runner. I can't definitively tell you what has gone wrong, but I can spot a few things that will be issues (there could also be others).

First, I think the log snippet here comes from the main log of the cloud runner. The underlying program error can often be found in another folder of the same run. You should see a "logs" folder and this should contain a program-specific log (like make_examples.log). This can be more informative.

From your run, I can see that you provide your FASTA as an uncompressed .FA file. I believe that the cloud runner requires a BGZIP compressed reference, a samtools FAIDX, and a GZI index all in the same place.

You will need to do the following operations on the FASTA file in a bucket that you control:

bgzip -i GRCh38_Verily_v1.genome.fa
samtools faidx GRCh38_Verily_v1.genome.fa

afterward, you will need to put the resulting:

GRCh38_Verily_v1.genome.fa.gz
GRCh38_Verily_v1.genome.fa.gz.gzi
GRCh38_Verily_v1.genome.fa.gz.fai

in the same bucket that the cloud runner can access. 

Hopefully these instructions seem reasonable and this unblocks you from this issue.

**Comment by saliksyed** [comment]
Okay great -- I will generate those files and add those using the '--ref_fai' and '--ref_gzi' flags respectively and let you know the results.

**Comment by saliksyed** [comment]
Ran it again, and still hitting failures:
```
done: true
error:
  code: 9
  message: 'Execution failed: action 1: unexpected exit status 1 was not ignored'
metadata:
  '@type': type.googleapis.com/google.genomics.v2alpha1.Metadata
  createTime: '2018-11-08T14:27:06.016940Z'
  endTime: '2018-11-08T14:30:59.324697Z'
  events:
  - description: Worker released
    details:
      '@type': type.googleapis.com/google.genomics.v2alpha1.WorkerReleasedEvent
      instance: google-pipelines-worker-4b16fd95b691baddc54b0c5ec50dc6c7
      zone: us-west1-b
    timestamp: '2018-11-08T14:30:59.324697Z'
  - description: 'Execution failed: action 1: unexpected exit status 1 was not ignored'
    details:
      '@type': type.googleapis.com/google.genomics.v2alpha1.FailedEvent
      cause: 'Execution failed: action 1: unexpected exit status 1 was not ignored'
      code: FAILED_PRECONDITION
    timestamp: '2018-11-08T14:30:58.518326Z'
  - description: Stopped running "/bin/sh -c gsutil -q cp /google/logs/output gs://canis/CNR-data/deep_variant_files/runner_logs_20181108_082705.log"
    details:
      '@type': type.googleapis.com/google.genomics.v2alpha1.ContainerStoppedEvent
      actionId: 2
      exitStatus: 0
      stderr: ''
    timestamp: '2018-11-08T14:30:58.416239Z'
  - description: Started running "/bin/sh -c gsutil -q cp /google/logs/output gs://canis/CNR-data/deep_variant_files/runner_logs_20181108_082705.log"
    details:
      '@type': type.googleapis.com/google.genomics.v2alpha1.ContainerStartedEvent
      actionId: 2
      ipAddress: ''
      portMappings: {}
    timestamp: '2018-11-08T14:30:55.929647Z'
  - description: Unexpected exit status 1 while running "-c /opt/deepvariant_runner/bin/gcp_deepvariant_runner   --project
      valis-194104   --zones us-west1-b   --docker_image gcr.io/deepvariant-docker/deepvariant:0.7.0   --outfile
      gs://canis/CNR-data/TLE_a_001_deep_variant.vcf   --staging gs://canis/CNR-data/deep_variant_files   --model
      gs://deepvariant/models/DeepVariant/0.7.0/DeepVariant-inception_v3-0.7.0+data-wes_standard   --regions
      gs://canis/CNR-data/CDS-canonical.bed   --bam gs://canis/CNR-data/TLE_a_001.bam   --bai
      gs://canis/CNR-data/TLE_a_001.bam.bai   --ref gs://canis/CNR-data/GRCh38_Verily_v1.genome.fa.gz    --ref_fai
      gs://canis/CNR-data/GRCh38_Verily_v1.genome.fa.gz.fai   --ref_gzi gs://canis/CNR-data/GRCh38_Verily_v1.genome.fa.gz.gzi   --gcsfuse"
    details:
      '@type': type.googleapis.com/google.genomics.v2alpha1.UnexpectedExitStatusEvent
      actionId: 1
      exitStatus: 1
    timestamp: '2018-11-08T14:30:55.434175Z'
  - description: |-
      Stopped running "-c /opt/deepvariant_runner/bin/gcp_deepvariant_runner   --project valis-194104   --zones us-west1-b   --docker_image gcr.io/deepvariant-docker/deepvariant:0.7.0   --outfile gs://canis/CNR-data/TLE_a_001_deep_variant.vcf   --staging gs://canis/CNR-data/deep_variant_files   --model gs://deepvariant/models/DeepVariant/0.7.0/DeepVariant-inception_v3-0.7.0+data-wes_standard   --regions gs://canis/CNR-data/CDS-canonical.bed   --bam gs://canis/CNR-data/TLE_a_001.bam   --bai gs://canis/CNR-data/TLE_a_001.bam.bai   --ref gs://canis/CNR-data/GRCh38_Verily_v1.genome.fa.gz    --ref_fai gs://canis/CNR-data/GRCh38_Verily_v1.genome.fa.gz.fai   --ref_gzi gs://canis/CNR-data/GRCh38_Verily_v1.genome.fa.gz.gzi   --gcsfuse": exit status 1: /examples_output.tfrecord@"${SHARDS}".gz\n     --reads "/input-gcsfused-{}/${BAM}"\n     --ref "${INPUT_REF}"\n     --task {}\n     --regions gs://canis/CNR-data/CDS-canonical.bed" # ENABLE_FUSE\n']
      Traceback (most recent call last):
        File "/opt/deepvariant_runner/src/gcp_deepvariant_runner.py", line 862, in <module>
          run()
        File "/opt/deepvariant_runner/src/gcp_deepvariant_runner.py", line 845, in run
          _run_make_examples(pipeline_args)
        File "/opt/deepvariant_runner/src/gcp_deepvariant_runner.py", line 340, in _run_make_examples
          _wait_for_results(threads, results)
        File "/opt/deepvariant_runner/src/gcp_deepvariant_runner.py", line 352, in _wait_for_results
          result.get()
        File "/usr/lib/python2.7/multiprocessing/pool.py", line 572, in get
          raise self._value
      RuntimeError: Job failed with error "run": operation "projects/valis-194104/operations/781565864516461293" failed: executing pipeline: Execution failed: action 6: unexpected exit status 1 was not ignored (reason: FAILED_PRECONDITION)
    details:
      '@type': type.googleapis.com/google.genomics.v2alpha1.ContainerStoppedEvent
      actionId: 1
      exitStatus: 1
      stderr: |+
        /examples_output.tfrecord@"${SHARDS}".gz\n     --reads "/input-gcsfused-{}/${BAM}"\n     --ref "${INPUT_REF}"\n     --task {}\n     --regions gs://canis/CNR-data/CDS-canonical.bed" # ENABLE_FUSE\n']
        Traceback (most recent call last):
          File "/opt/deepvariant_runner/src/gcp_deepvariant_runner.py", line 862, in <module>
            run()
          File "/opt/deepvariant_runner/src/gcp_deepvariant_runner.py", line 845, in run
            _run_make_examples(pipeline_args)
          File "/opt/deepvariant_runner/src/gcp_deepvariant_runner.py", line 340, in _run_make_examples
            _wait_for_results(threads, results)
          File "/opt/deepvariant_runner/src/gcp_deepvariant_runner.py", line 352, in _wait_for_results
            result.get()
          File "/usr/lib/python2.7/multiprocessing/pool.py", line 572, in get
            raise self._value
        RuntimeError: Job failed with error "run": operation "projects/valis-194104/operations/781565864516461293" failed: executing pipeline: Execution failed: action 6: unexpected exit status 1 was not ignored (reason: FAILED_PRECONDITION)

    timestamp: '2018-11-08T14:30:55.372655Z'
  - description: Started running "-c /opt/deepvariant_runner/bin/gcp_deepvariant_runner   --project
      valis-194104   --zones us-west1-b   --docker_image gcr.io/deepvariant-docker/deepvariant:0.7.0   --outfile
      gs://canis/CNR-data/TLE_a_001_deep_variant.vcf   --staging gs://canis/CNR-data/deep_variant_files   --model
      gs://deepvariant/models/DeepVariant/0.7.0/DeepVariant-inception_v3-0.7.0+data-wes_standard   --regions
      gs://canis/CNR-data/CDS-canonical.bed   --bam gs://canis/CNR-data/TLE_a_001.bam   --bai
      gs://canis/CNR-data/TLE_a_001.bam.bai   --ref gs://canis/CNR-data/GRCh38_Verily_v1.genome.fa.gz    --ref_fai
      gs://canis/CNR-data/GRCh38_Verily_v1.genome.fa.gz.fai   --ref_gzi gs://canis/CNR-data/GRCh38_Verily_v1.genome.fa.gz.gzi   --gcsfuse"
    details:
      '@type': type.googleapis.com/google.genomics.v2alpha1.ContainerStartedEvent
      actionId: 1
      ipAddress: ''
      portMappings: {}
    timestamp: '2018-11-08T14:28:16.559791Z'
  - description: Stopped pulling "google/cloud-sdk:alpine"
    details:
      '@type': type.googleapis.com/google.genomics.v2alpha1.PullStoppedEvent
      imageUri: google/cloud-sdk:alpine
    timestamp: '2018-11-08T14:28:12.427407Z'
  - description: Started pulling "google/cloud-sdk:alpine"
    details:
      '@type': type.googleapis.com/google.genomics.v2alpha1.PullStartedEvent
      imageUri: google/cloud-sdk:alpine
    timestamp: '2018-11-08T14:28:05.897359Z'
  - description: Stopped pulling "gcr.io/deepvariant-docker/deepvariant_runner:0.7.0"
    details:
      '@type': type.googleapis.com/google.genomics.v2alpha1.PullStoppedEvent
      imageUri: gcr.io/deepvariant-docker/deepvariant_runner:0.7.0
    timestamp: '2018-11-08T14:28:05.747135Z'
  - description: Started pulling "gcr.io/deepvariant-docker/deepvariant_runner:0.7.0"
    details:
      '@type': type.googleapis.com/google.genomics.v2alpha1.PullStartedEvent
      imageUri: gcr.io/deepvariant-docker/deepvariant_runner:0.7.0
    timestamp: '2018-11-08T14:27:34.961215Z'
  - description: Worker "google-pipelines-worker-4b16fd95b691baddc54b0c5ec50dc6c7"
      assigned in "us-west1-b"
    details:
      '@type': type.googleapis.com/google.genomics.v2alpha1.WorkerAssignedEvent
      instance: google-pipelines-worker-4b16fd95b691baddc54b0c5ec50dc6c7
      zone: us-west1-b
    timestamp: '2018-11-08T14:27:06.604193Z'
  labels: {}
  pipeline:
    actions:
    - commands:
      - -c
      - /opt/deepvariant_runner/bin/gcp_deepvariant_runner   --project valis-194104   --zones
        us-west1-b   --docker_image gcr.io/deepvariant-docker/deepvariant:0.7.0   --outfile
        gs://canis/CNR-data/TLE_a_001_deep_variant.vcf   --staging gs://canis/CNR-data/deep_variant_files   --model
        gs://deepvariant/models/DeepVariant/0.7.0/DeepVariant-inception_v3-0.7.0+data-wes_standard   --regions
        gs://canis/CNR-data/CDS-canonical.bed   --bam gs://canis/CNR-data/TLE_a_001.bam   --bai
        gs://canis/CNR-data/TLE_a_001.bam.bai   --ref gs://canis/CNR-data/GRCh38_Verily_v1.genome.fa.gz    --ref_fai
        gs://canis/CNR-data/GRCh38_Verily_v1.genome.fa.gz.fai   --ref_gzi gs://canis/CNR-data/GRCh38_Verily_v1.genome.fa.gz.gzi   --gcsfuse
      entrypoint: bash
      environment: {}
      flags: []
      imageUri: gcr.io/deepvariant-docker/deepvariant_runner:0.7.0
      labels: {}
      mounts: []
      name: ''
      pidNamespace: ''
      portMappings: {}
    - commands:
      - /bin/sh
      - -c
      - gsutil -q cp /google/logs/output gs://canis/CNR-data/deep_variant_files/runner_logs_20181108_082705.log
      entrypoint: ''
      environment: {}
      flags:
      - ALWAYS_RUN
      imageUri: google/cloud-sdk:alpine
      labels: {}
      mounts: []
      name: ''
      pidNamespace: ''
      portMappings: {}
    environment: {}
    resources:
      projectId: valis-194104
      regions: []
      virtualMachine:
        accelerators: []
        bootDiskSizeGb: 10
        bootImage: projects/cos-cloud/global/images/family/cos-stable
        cpuPlatform: ''
        disks: []
        labels: {}
        machineType: n1-standard-1
        nvidiaDriverVersion: ''
        preemptible: false
        serviceAccount:
          email: default
          scopes:
          - https://www.googleapis.com/auth/cloud-platform
          - https://www.googleapis.com/auth/devstorage.read_write
          - https://www.googleapis.com/auth/genomics
      zones:
      - us-west1-b
    timeout: 604800s
  startTime: '2018-11-08T14:27:06.604193Z'
name: projects/valis-194104/operations/12097970745380060156
```

**Comment by pichuan** [comment]
+ @nmousavi  from the Google Cloud team to take a look.
I'll also take a look later today when I find some more time.

**Comment by nmousavi** [comment]
As @AndrewCarroll said, we need workers log to investigate the failure. Please share workers log (you should be able to find them under `gs://canis/CNR-data/deep_variant_files/logs`).

**Comment by pgrosu** [comment]
Actually the answer is right in front of you - you don't need the logs.  The prize goes to the first one who sees it :)

**Comment by saliksyed** [comment]
Lack of job name prefix?
https://github.com/google/deepvariant/blob/r0.7/deepvariant/docker/gcp_deepvariant_runner.py#L845

**Comment by pgrosu** [comment]
@saliksyed Try replacing `google/cloud-sdk:alpine` with `google/cloud-sdk:slim`, and then run it again to see if that fixes it.

Thanks,
~p

**Comment by saliksyed** [comment]
I'm not sure how to specify the gcloud sdk version in the launch script -- is there an option for that?

**Comment by pgrosu** [comment]
The Docker container for the Google pipeline backend would need to be fixed.  Take a look, I can cause an exit status of 1 with `slim` and `alpine`, but I can only get a clean exit from `slim` - basically it's getting stuck with `alpine` and not exiting:

```
paul$ sudo docker run google/cloud-sdk:slim gsutil -m cp /not/there gs://my-bucket/
CommandException: No URLs matched: /not/there
CommandException: 1 file/object could not be transferred.
paul$ sudo docker run google/cloud-sdk:alpine gsutil -m cp /not/there gs://my-bucket/
CommandException: No URLs matched: /not/there
CommandException: 1 file/object could not be transferred.


```

Hope it helps,
~p

**Comment by nmousavi** [comment]
Again, please share workers log so we can help.

**Comment by saliksyed** [comment]
@nmousavi I made a readonly public bucket you can access here:
http://storage.googleapis.com/public-debug

@pgrosu I am not using Docker. I am running the shell script from my local machine. Do I need to update my google-cloud-sdk version to slim? Or are you referring to the DeepVariant container image itself? If so, I am using the Cloud Genomics API -- not hosting my own image of DeepVariant... see:
https://cloud.google.com/genomics/docs/tutorials/deepvariant

**Comment by nmousavi** [comment]
It has nothing to do with docker image. The problem is with input file (with err `Unrecognized SAM header type`). Pasting make_example worker log:


```
Unrecognized SAM header type, ignoring:
I1108 21:36:46.455516 140363989006080 genomics_reader.py:213] Reading /input-gcsfused-2/CNR-data/TLE_a_001.bam with NativeSamReader
Traceback (most recent call last):
  File "/mnt/google/.google/tmp/Bazel.runfiles_pV6vDu/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 1188, in <module>
    tf.app.run()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/platform/app.py", line 125, in run
    _sys.exit(main(argv))
  File "/mnt/google/.google/tmp/Bazel.runfiles_pV6vDu/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 1178, in main
    make_examples_runner(options)
  File "/mnt/google/.google/tmp/Bazel.runfiles_pV6vDu/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 1076, in make_examples_runner
    regions = processing_regions_from_options(options)
  File "/mnt/google/.google/tmp/Bazel.runfiles_pV6vDu/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 990, in processing_regions_from_options
    options.min_shared_contigs_basepairs)
  File "/mnt/google/.google/tmp/Bazel.runfiles_pV6vDu/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 485, in _ensure_consistent_contigs
    min_coverage_fraction)
  File "/mnt/google/.google/tmp/Bazel.runfiles_pV6vDu/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 557, in validate_reference_contig_coverage
    ref_bp, common_bp, coverage, format_contig_matches()))
ValueError: Reference contigs span 3088269832 bases but only 0 bases (0.00%) were found in common among our input files. Check that the sources were created on a common genome reference build. Contig matches were: "chr1" is 248956422 bp and IS MISSING, "chr2" is 242193529 bp and IS MISSING, "chr3" is 198295559 bp and IS MISSING, "chr4" is 190214555 bp and IS MISSING, "chr5" is 181538259 bp and IS MISSING, "chr6" is 170805979 bp and IS MISSING, "chr7" is 159345973 bp and IS MISSING, "chr8" is 145138636 bp and IS MISSING, "chr9" is 138394717 bp and IS MISSING, "chr10" is 133797422 bp and IS MISSING, "chr11" is 135086622 bp and IS MISSING, "chr12" is 133275309 bp and IS MISSING, "chr13" is 114364328 bp and IS MISSING, "chr14" is 107043718 bp and IS MISSING, "chr15" is 101991189 bp and IS MISSING, "chr16" is 90338345 bp and IS MISSING, "chr17" is 83257441 bp and IS MISSING, "chr18" is 80373285 bp and IS MISSING, "chr19" is 58617616 bp and IS MISSING, "chr20" is 64444167 bp and IS MISSING, "chr21" is 46709983 bp and IS MISSING, "chr22" is 50818468 bp and IS MISSING, "chrX" is 156040895 bp and IS MISSING, "chrY" is 57227415 bp and IS MISSING
parallel: This job failed:
mkdir -p ./input-gcsfused-2 && gcsfuse --implicit-dirs canis /input-gcsfused-2 && /opt/deepvariant/bin/make_examples --mode calling --examples /mnt/google/.google/output/canis/CNR-data/deep_variant_files/examples/0/examples_output.tfrecord@8.gz --reads /input-gcsfused-2/CNR-data/TLE_a_001.bam --ref /mnt/google/.google/input/canis/CNR-data/GRCh38_Verily_v1.genome.fa.gz --task 2 --regions gs://canis/CNR-data/exomes.bed
Using mount point: /input-gcsfused-2

```

Assigning it to Pi-Chuan.

**Comment by pgrosu** [comment]
@saliksyed  The real issue is that your BAM file does not contain those chromosomes/contigs as based on the reference file.  That would probably will fix it.  Sorry for the confusion, as the error was giving me a different signal.

@nmousavi Not exactly, but I understand what's the trigger, and it probably would be helpful to provide more direct causes directly in the description of the events.

The real thing that's happening here is that he doesn't have the right chromosome labels in his BAM files as compared to the reference genome, which misses all of them and triggers the error on the following lines of code:

https://github.com/google/deepvariant/blob/r0.7/deepvariant/make_examples.py#L556-L561

The `Unrecognized SAM header type, ignoring:` is just the following warning-line of code that doesn't trigger the exit status of 1:

https://github.com/google/deepvariant/blob/aba55537eec832e6cea678349422124ef50680f4/third_party/nucleus/io/sam_reader.cc#L525

Thanks,
~p

**Comment by saliksyed** [comment]
Okay so I get the following information, which indicates the reference genome used for alignment is GRCh37 (not 38 as was assumed). So looks like I need to get a different reference genome. 
```
@SQ     SN:chr1 LN:249250621
@SQ     SN:chr2 LN:243199373
@SQ     SN:chr3 LN:198022430
@SQ     SN:chr4 LN:191154276
@SQ     SN:chr5 LN:180915260
@SQ     SN:chr6 LN:171115067
@SQ     SN:chr7 LN:159138663
@SQ     SN:chr8 LN:146364022
@SQ     SN:chr9 LN:141213431
@SQ     SN:chr10        LN:135534747
@SQ     SN:chr11        LN:135006516
@SQ     SN:chr12        LN:133851895
@SQ     SN:chr13        LN:115169878
@SQ     SN:chr14        LN:107349540
@SQ     SN:chr15        LN:102531392
@SQ     SN:chr16        LN:90354753
@SQ     SN:chr17        LN:81195210
@SQ     SN:chr18        LN:78077248
@SQ     SN:chr19        LN:59128983
@SQ     SN:chr20        LN:63025520
@SQ     SN:chr21        LN:48129895
@SQ     SN:chr22        LN:51304566
@SQ     SN:chrX LN:155270560
@SQ     SN:chrY LN:59373566
@SQ     SN:chrM LN:16569
```

**Comment by saliksyed** [comment]
Thank you also to @pgrosu for taking a look. I think the next steps for me are:
1 -- disassemble the BAM file and re-align using BWA-MEM to the Verily Genome
2 -- Re run the pipeline with the GRCh38 aligned file

**Comment by depristo** [comment]
Why not just run the pipeline using the b37 version of the genome? The gcp runner pipeline should support any reference genome

**Comment by pichuan** [comment]
@saliksyed the example on the Cloud runner page is actually a b37 genome:

```
  --ref gs://deepvariant/performance-testdata/hs37d5.fa.gz \
```

See if that works for you? It should also already come with the corresponding indexed file and ready to go.

**Comment by saliksyed** [comment]
@depristo : we are correlating the outputted VCF against a bunch of data that is in GRCh38. AFAIK you can't convert the GRCh37 vcf into GRCh38 easily right?

**Comment by saliksyed** [comment]
I ran using the reference genome mentioned earlier, but it seems that the reference has the chromosomes named without the "chr" prefix, which the BAM file has for the contig names, it would be nice if the package could detect correct this automatically -- do you think this makes sense, if so I am happy to submit a patch:
```
2018-11-10 01:31:37.340729: W third_party/nucleus/io/sam_reader.cc:537] Unrecognized SAM header type, ignoring: 
I1110 01:31:37.359072 140105147934464 genomics_reader.py:213] Reading /input-gcsfused-0/CNR-data/TLE_a_001.bam with NativeSamReader
I1110 01:31:37.383352 140105147934464 make_examples.py:1075] Preparing inputs
2018-11-10 01:31:38.214366: W third_party/nucleus/io/sam_reader.cc:537] Unrecognized SAM header type, ignoring: 
I1110 01:31:38.223048 140105147934464 genomics_reader.py:213] Reading /input-gcsfused-0/CNR-data/TLE_a_001.bam with NativeSamReader
Traceback (most recent call last):
  File "/mnt/google/.google/tmp/Bazel.runfiles_Mu9e5m/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 1188, in <module>
    tf.app.run()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/platform/app.py", line 125, in run
    _sys.exit(main(argv))
  File "/mnt/google/.google/tmp/Bazel.runfiles_Mu9e5m/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 1178, in main
    make_examples_runner(options)
  File "/mnt/google/.google/tmp/Bazel.runfiles_Mu9e5m/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 1076, in make_examples_runner
    regions = processing_regions_from_options(options)
  File "/mnt/google/.google/tmp/Bazel.runfiles_Mu9e5m/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 990, in processing_regions_from_options
    options.min_shared_contigs_basepairs)
  File "/mnt/google/.google/tmp/Bazel.runfiles_Mu9e5m/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 485, in _ensure_consistent_contigs
    min_coverage_fraction)
  File "/mnt/google/.google/tmp/Bazel.runfiles_Mu9e5m/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 557, in validate_reference_contig_coverage
    ref_bp, common_bp, coverage, format_contig_matches()))
ValueError: Reference contigs span 3095677412 bases but only 0 bases (0.00%) were found in common among our input files. Check that the sources were created on a common genome reference build. Contig matches were: "1" is 249250621 bp and IS MISSING, "2" is 243199373 bp and IS MISSING, "3" is 198022430 bp and IS MISSING, "4" is 191154276 bp and IS MISSING, "5" is 180915260 bp and IS MISSING, "6" is 171115067 bp and IS MISSING, "7" is 159138663 bp and IS MISSING, "8" is 146364022 bp and IS MISSING, "9" is 141213431 bp and IS MISSING, "10" is 135534747 bp and IS MISSING, "11" is 135006516 bp and IS MISSING, "12" is 133851895 bp and IS MISSING, "13" is 115169878 bp and IS MISSING, "14" is 107349540 bp and IS MISSING, "15" is 102531392 bp and IS MISSING, "16" is 90354753 bp and IS MISSING, "17" is 81195210 bp and IS MISSING, "18" is 78077248 bp and IS MISSING, "19" is 59128983 bp and IS MISSING, "20" is 63025520 bp and IS MISSING, "21" is 48129895 bp and IS MISSING, "22" is 51304566 bp and IS MISSING, "X" is 155270560 bp and IS MISSING, "Y" is 59373566 bp and IS MISSING
parallel: This job failed:
mkdir -p ./input-gcsfused-0 && gcsfuse --implicit-dirs canis /input-gcsfused-0 && /opt/deepvariant/bin/make_examples --mode calling --examples /mnt/google/.google/output/canis/CNR-data/deep_variant_files/examples/0/examples_output.tfrecord@8.gz --reads /input-gcsfused-0/CNR-data/TLE_a_001.bam --ref /mnt/google/.google/input/deepvariant/performance-testdata/hs37d5.fa.gz --task 0 --regions gs://canis/CNR-data/exomes.bed
Using mount point: /input-gcsfused-0
Opening GCS connection...
Opening bucket...
Mounting file system...
File system has been successfully mounted.
```

**Comment by saliksyed** [comment]
I was able to re-header the BAM file using samtools, now the script fails at a new step: reading the BED file. I am not sure what the problem is here:
```
Using mount point: /input-gcsfused-0
Opening GCS connection...
Opening bucket...
Mounting file system...
File system has been successfully mounted.
2018-11-11 17:32:17.333150: W third_party/nucleus/io/sam_reader.cc:537] Unrecognized SAM header type, ignoring: 
I1111 17:32:17.342442 140289241822976 genomics_reader.py:213] Reading /input-gcsfused-0/CNR-data/TLE_a_001.reheader.bam with NativeSamReader
I1111 17:32:17.361879 140289241822976 make_examples.py:1075] Preparing inputs
2018-11-11 17:32:18.174362: W third_party/nucleus/io/sam_reader.cc:537] Unrecognized SAM header type, ignoring: 
I1111 17:32:18.184726 140289241822976 genomics_reader.py:213] Reading /input-gcsfused-0/CNR-data/TLE_a_001.reheader.bam with NativeSamReader
I1111 17:32:18.227871 140289241822976 make_examples.py:991] Common contigs are [u'1', u'2', u'3', u'4', u'5', u'6', u'7', u'8', u'9', u'10', u'11', u'12', u'13', u'14', u'15', u'16', u'17', u'18', u'19', u'20', u'21', u'22', u'X', u'Y']
[E::hts_open_format] Failed to open file gs://canis/CNR-data/exomes.bed
Traceback (most recent call last):
  File "/mnt/google/.google/tmp/Bazel.runfiles_34p88R/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 1188, in <module>
    tf.app.run()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/platform/app.py", line 125, in run
    _sys.exit(main(argv))
  File "/mnt/google/.google/tmp/Bazel.runfiles_34p88R/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 1178, in main
    make_examples_runner(options)
  File "/mnt/google/.google/tmp/Bazel.runfiles_34p88R/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 1076, in make_examples_runner
    regions = processing_regions_from_options(options)
  File "/mnt/google/.google/tmp/Bazel.runfiles_34p88R/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 993, in processing_regions_from_options
    options.exclude_calling_regions)
  File "/mnt/google/.google/tmp/Bazel.runfiles_34p88R/runfiles/com_google_deepvariant/deepvariant/make_examples.py", line 587, in build_calling_regions
    ranges.RangeSet.from_regions(regions_to_include, contig_dict))
  File "/mnt/google/.google/tmp/Bazel.runfiles_34p88R/runfiles/com_google_deepvariant/third_party/nucleus/util/ranges.py", line 161, in from_regions
    return cls(ranges=from_regions(regions, contig_map=contig_map))
  File "/mnt/google/.google/tmp/Bazel.runfiles_34p88R/runfiles/com_google_deepvariant/third_party/nucleus/util/ranges.py", line 113, in __init__
    for i, range_ in enumerate(ranges):
  File "/mnt/google/.google/tmp/Bazel.runfiles_34p88R/runfiles/com_google_deepvariant/third_party/nucleus/util/ranges.py", line 464, in from_regions
    for elt in reader(region):
  File "/mnt/google/.google/tmp/Bazel.runfiles_34p88R/runfiles/com_google_deepvariant/third_party/nucleus/util/ranges.py", line 429, in bed_parser
    with bed.BedReader(filename) as fin:
  File "/mnt/google/.google/tmp/Bazel.runfiles_34p88R/runfiles/com_google_deepvariant/third_party/nucleus/io/genomics_reader.py", line 211, in __init__
    self._reader = self._native_reader(input_path, **kwargs)
  File "/mnt/google/.google/tmp/Bazel.runfiles_34p88R/runfiles/com_google_deepvariant/third_party/nucleus/io/bed.py", line 127, in _native_reader
    return NativeBedReader(input_path, **kwargs)
  File "/mnt/google/.google/tmp/Bazel.runfiles_34p88R/runfiles/com_google_deepvariant/third_party/nucleus/io/bed.py", line 104, in __init__
    self._reader = bed_reader.BedReader.from_file(bed_path, options)
ValueError: Not found: Could not open gs://canis/CNR-data/exomes.bed
parallel: This job failed:
mkdir -p ./input-gcsfused-0 && gcsfuse --implicit-dirs canis /input-gcsfused-0 && /opt/deepvariant/bin/make_examples --mode calling --examples /mnt/google/.google/output/canis/CNR-data/deep_variant_files/examples/0/examples_output.tfrecord@8.gz --reads /input-gcsfused-0/CNR-data/TLE_a_001.reheader.bam --ref /mnt/google/.google/input/deepvariant/performance-testdata/hs37d5.fa.gz --task 0 --regions gs://canis/CNR-data/exomes.bed
```

I've added the BED file to the public bucket:
gs://public-debug/exomes.bed

**Comment by pichuan** [comment]
First of all, I assume that you have access to `gs://canis/CNR-data/exomes.bed` file?
If you do:
`gsutil cat gs://canis/CNR-data/exomes.bed` from the project that has the same permission, you can first double check that you have access to it.

And, I don't fully understand how gcsfuse is implemented in the Google Cloud runner, so I'll defer this to our collaborator at Cloud @nmousavi  -- Nima, in the command in the previous log, particularly this one:
`mkdir -p ./input-gcsfused-0 && gcsfuse --implicit-dirs canis /input-gcsfused-0 && /opt/deepvariant/bin/make_examples --mode calling --examples /mnt/google/.google/output/canis/CNR-data/deep_variant_files/examples/0/examples_output.tfrecord@8.gz --reads /input-gcsfused-0/CNR-data/TLE_a_001.reheader.bam --ref /mnt/google/.google/input/deepvariant/performance-testdata/hs37d5.fa.gz --task 0 --regions gs://canis/CNR-data/exomes.bed`

It seems weird to me that everything else has been changed to under `/input-gcsfused-0` or `/mnt`, but the BED file still has a `gs://` prefix. This seems to me like something might be wrong in how the runner is using gcsfuse on the BED file. Can you take a look?

Thanks!

**Comment by saliksyed** [comment]
@pichuan Thank you for the followup. I definitely have access to the exomes.bed file and the gsutil cat works fine. 

I see what you mean re: the `/input-gcsfused-0/CNR-data/` ... I did not notice this change and assumed I could just use the gs:// format of the input. That is a good observation! I will try it with this change.

**Comment by saliksyed** [comment]
Seems this issues is a duplicate of https://github.com/google/deepvariant/issues/118

**Comment by nmousavi** [comment]
I was able to run a test pipeline successfully with BED file you provided (`gs://public-debug/exomes.bed`). Could you please run your pipeline with `gs://public-debug/exomes.bed` (not `gs://canis/CNR-data/exomes.bed`) and see if it succeeds.


This is what I got in the log

```
I1112 17:29:38.624562 140623242430208 genomics_reader.py:213] Reading gs://public-debug/exomes.bed with NativeBedReader
I1112 17:30:18.942625 140623242430208 make_examples.py:1086] Writing examples to /mnt/google/.google/output/nmousavi-test/dv-test/2018-11-10/staging/examples/0/examples_output.tfrecord-00000-of-00008.gz

```

**Comment by saliksyed** [comment]
Okay, I will try this with my exomes.bed file. I was already able to successfully run the pipeline with the following params
```
  --regions gs://deepvariant/exome-case-study-testdata/refseq.coding_exons.b37.extended50.bed \
  --bam gs://canis/CNR-data/TLE_a_001.reheader.bam \
  --bai gs://canis/CNR-data/TLE_a_001.reheader.bam.bai \
  --ref gs://deepvariant/performance-testdata/hs37d5.fa.gz \
```

It is odd that the bed file needs to lie in a public bucket, while the genomic data does not.

**Comment by nmousavi** [comment]
Lets first see if that's the case (i.e. having bed file in public bucket resolves it). If yes, this is a bug and we will fix it.

**Comment by saliksyed** [comment]
I can confirm that placing the exomes.bed file in `gs://public-debug` results in successful reading of the BED file (though the pipeline fails at a latter step due to inconsistent contig names within the BED vs. the reference genome).

**Comment by saliksyed** [comment]
Thanks for all the help @nmousavi , we can certainly use a public bucket for the region files as there is no PII.

**Comment by pichuan** [comment]
@saliksyed 
Hopefully this is resolved with v0.7.2 release, and with the workaround of putting the BED file in a public bucket.
If you encounter more issues, let us know.
